src/main/java/racingcar/domain/Cars.java

```diff
@@ -0,0 +1,60 @@
+package racingcar.domain;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import racingcar.dto.CarDto;
+
+public class Cars {
+    private static final String ERROR_DUPLICATE_NAME = "[ERROR] 동일한 이름은 입력할 수 없습니다. 다시 입력해주세요.";
+    private static final String ERROR_NOT_FIND_MAX_POSITION = "[ERROR] 최대 거리를 찾을 수 없습니다.";
+
+    private final List<Car> cars;
+
+    private Cars(List<Car> cars) {
+        this.cars = cars;
+    }
+
+    public static Cars from(List<String> carNames) {
+        validateDuplicate(carNames);
+        List<Car> cars = carNames.stream()
+            .map(Car::new)
+            .collect(Collectors.toList());
+        return new Cars(cars);
+    }
+
+    private static void validateDuplicate(List<String> carNames) {
+        Set<String> nonDuplicateNames = new HashSet<>(carNames);
+        if (nonDuplicateNames.size() != carNames.size()) {
+            throw new IllegalArgumentException(ERROR_DUPLICATE_NAME);
+        }
+    }
+
+    public void move() {
+        cars.forEach(Car::move);
```

Comment:
```suggestion
        for (Car car : cars) {
            car.move();
        }
```

[Iterable.forEach() 와 for-each 루프의 차이](https://newbedev.com/java-8-iterable-foreach-vs-foreach-loop)에 대해 알아보면 좋겠어요. 🙃

Reply:
말씀하신 링크가 영어라 잘 와닿지가 않아 검색해보았더니 
https://tecoble.techcourse.co.kr/post/2020-05-14-foreach-vs-forloop/
이런 글이 나왔습니다. 
스트림의 forEach 구문과 For-each 루프에 대한 내용이지만 동일하다고 봐도 될까요?

Reply:
네, 동일하다고 생각하고 보셔도 큰 문제 없을 것 같아요! 👍